Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,235 @@
+/-
+Copyright (c) 2024 Colin Jones. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Colin Jones
+-/
+import Mathlib.NumberTheory.Divisors
+import Mathlib.Tactic.NormNum.Prime
+import Mathlib.Algebra.GeomSum
+import Mathlib.Tactic.FinCases
+
+/-!
+## Factorisation properties of natural numbers
+This file defines abundant, pseudoperfect, deficient, and weird numbers and formalizes their
+  relations with prime and perfect numbers.
+
+## Main Definitions
+* `Nat.Abundant`: a natural number `n` is abundant if the sum of its proper divisors are greater
+  than `n`
+* `Nat.Pseudoperfect`: a natural number `n` is pseudoperfect if a subset of its proper divisors
+  equals `n`
+* `Nat.Deficient`: a natural number `n` is deficient if the sum of its proper divisors is less than
+  `n`
+* `Nat.WeirdNumber`: a natural number is weird if it is both abundant and not pseudoperfect
+
+## Main Results
+* `Nat.deficient_or_perfect_or_abundant`: a positive natural number is either deficient, perfect, or
+  abundant
+* `Nat.prime_deficient`: all prime natural numbers are deficient
+* `Nat.exists_infinite_deficients`: there are infinitely many deficient numbers
+* `Nat.prime_pow_deficient`: any natural number power of a prime is deficient
+
+## Implementation Details
+* Zero is not included in any of the definitions and these definitions only apply to natural
+  numbers greater than zero.
+
+## References
+* Prielipp, Robert W. “PERFECT NUMBERS, ABUNDANT NUMBERS, AND DEFICIENT NUMBERS.”
+  The Mathematics Teacher, vol. 63, no. 8, 1970, pp. 692–96. JSTOR,
+  http://www.jstor.org/stable/27958492. Accessed 21 Feb. 2024.
+
+## Tags
+
+abundant, deficient, weird, pseudoperfect
+-/
+
+open Nat BigOperators Finset
+
+namespace Nat
+
+variable (n m k : ℕ)
```

Comment:
`k` never gets used. Moreover, for most of the lemmas, `n` can be inferred from the other arguments so it doesn't need to be given explicitly; that is, you should do `variable {n : ℕ}` and then override it with `(n : ℕ)` in individual declarations where it isn't inferrable. (Note that in theorems whose conclusion is an "iff" statement, it's usual to have variables implicit if they can be inferred from both sides of the "iff", which is the case for your lemma `not_pseudoperfect_iff_forall` for example.)

Reply:
The `k` was an artifact that I did not catch. I just removed the `variable (n : \N)` part of the code. Is that okay?